From ad110a2 ICU-3350 Compute the WEEK_OF_MONTH shift for any Gregorian cutover
From: Dan Yeaw <dan@yeaw.me>
Date: Tue Oct 27 2026 00:00:00 +0000
Subject: [PATCH] Use python instead of py launcher on Windows

ICU's Windows makefiles invoke the Windows "py" launcher ("py -3 ..."),
which is not always installed (e.g. Microsoft Store Python does not
include py.exe). Fall back to plain "python" so the build works with any
Python on PATH.
---
 source/data/makedata.mak | 2 +-
 source/test/testdata/testdata.mak | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/source/data/makedata.mak b/source/data/makedata.mak
index 4bf8485..eaac6e1 100644
--- a/source/data/makedata.mak
+++ b/source/data/makedata.mak
@@ -33,7 +33,7 @@ ICU_LIB_TARGET=$(DLL_OUTPUT)$(U_ICUDATA_NAME).dll
 !ENDIF
 !MESSAGE ICU data make path is $(ICUMAKE)
 
-!IF [py -3 -c "exit(0)"]!=0
+!IF [python -c "exit(0)"]!=0
 !MESSAGE Information: Unable to find Python 3. Data will fail to build from source.
 !ENDIF
 
@@ -305,7 +305,7 @@ $(TOOLS_TS): "$(ICUTOOLS)\genrb\$(CFGTOOLS)\genrb.exe" "$(ICUTOOLS)\gencnval\$(C
 $(COREDATA_TS):
 	@cd "$(ICUSRCDATA)"
 	set PYTHONPATH=$(ICUP)\source\python;%PYTHONPATH%
-	py -3 -B -m icutools.databuilder \
+	python -B -m icutools.databuilder \
 		--mode windows-exec \
 		--src_dir "$(ICUSRCDATA)" \
 		--tool_dir "$(ICUTOOLS)" \
diff --git a/source/test/testdata/testdata.mak b/source/test/testdata/testdata.mak
index 2809efd..c20ea05 100644
--- a/source/test/testdata/testdata.mak
+++ b/source/test/testdata/testdata.mak
@@ -35,7 +35,7 @@ CREATE_DIRS :
 "$(TESTDATAOUT)\testdata.dat" :
 	@echo Building test data
 	set PYTHONPATH=$(ICUP)\source\python;%PYTHONPATH%
-	py -3 -B -m icutools.databuilder \
+	python -B -m icutools.databuilder \
 		--mode windows-exec \
 		--tool_dir "$(ICUTOOLS)" \
 		--tool_cfg "$(CFG)" \
